subprojects/diagnostics/src/main/groovy/org/gradle/api/tasks/diagnostics/BuildscriptDependencyReportTask.java

```diff
@@ -0,0 +1,143 @@
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.diagnostics.internal.DependencyReportRenderer;
+import org.gradle.api.tasks.diagnostics.internal.ProjectReportGenerator;
+import org.gradle.api.tasks.diagnostics.internal.ReportGenerator;
+import org.gradle.api.tasks.diagnostics.internal.ReportRenderer;
+import org.gradle.api.tasks.diagnostics.internal.dependencies.AsciiDependencyReportRenderer;
+import org.gradle.initialization.BuildClientMetaData;
+import org.gradle.logging.StyledTextOutputFactory;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * Displays the buildscript dependency tree for a project. An instance of this type is used when you
+ * execute the {@code buildscriptDependencies} task from the command-line.
+ */
+public class BuildscriptDependencyReportTask extends ConventionTask {
+
+    private DependencyReportRenderer renderer = new AsciiDependencyReportRenderer();
+    private File outputFile;
+    private Set<Project> projects;
+
+    public BuildscriptDependencyReportTask() {
+        getOutputs().upToDateWhen(new Spec<Task>() {
+            public boolean isSatisfiedBy(Task element) {
+                return false;
+            }
+        });
+        projects = new HashSet<Project>();
+        projects.add(getProject());
+    }
+
+    @Inject
+    protected BuildClientMetaData getClientMetaData() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Inject
+    protected StyledTextOutputFactory getTextOutputFactory() {
+        throw new UnsupportedOperationException();
+    }
+
+    @TaskAction
+    public void generate() {
+        ProjectReportGenerator projectReportGenerator = new ProjectReportGenerator() {
+            @Override
+            public void generateReport(Project project) throws IOException {
+                SortedSet<Configuration> sortedConfigurations = new TreeSet<Configuration>(new Comparator<Configuration>() {
+                    public int compare(Configuration conf1, Configuration conf2) {
+                        return conf1.getName().compareTo(conf2.getName());
+                    }
+                });
+                sortedConfigurations.addAll(getProject().getBuildscript().getConfigurations());
+                for (Configuration configuration : sortedConfigurations) {
+                    renderer.startConfiguration(configuration);
+                    renderer.render(configuration);
+                    renderer.completeConfiguration(configuration);
+                }
+            }
+        };
+
+        ReportGenerator reportGenerator = new ReportGenerator(getRenderer(), getClientMetaData(), getOutputFile(),
+                getTextOutputFactory(), projectReportGenerator);
+        reportGenerator.generateReport(new TreeSet<Project>(getProjects()));
+    }
+
+    public ReportRenderer getRenderer() {
```

Comment:
We'll need to remove all of these getters and setters for the first iteration, and later find a better way. That is, the initial version will be hardcoded to use the ascii renderer and render to the console. 

The main reason for this is that some of these types are not public while this task type is. Also, we need to not expose any coupling to the existing reporting structure at this time. This also includes hardcoding to apply to the project that the task is added to.